library/src/engine.cxx

```diff
@@ -97,6 +101,72 @@ interactor& engine::getInteractor()
   return *this->Internals->Interactor;
 }
 
+//----------------------------------------------------------------------------
+void engine::loadPlugin(const std::string& path)
+{
+  // check if the plugin is a known static plugin
+  f3d::factory* factory = f3d::factory::instance();
+  factory::plugin_initializer_t init_plugin = factory->getStaticInitializer(path);
+
+  if (init_plugin == nullptr)
+  {
+    // try if the path is a library full path
+    vtksys::DynamicLoader::LibraryHandle handle = vtksys::DynamicLoader::OpenLibrary(path);
+    if (!handle)
+    {
+      std::string fullPath;
+
+      // Right now, plugins should be located in the same folder than f3d.exe on Windows
+      // On Linux/macOS, when we are not using a full path, so we rely on LD_LIBRARY_PATH
+#ifdef _WIN32
+      std::vector<wchar_t> pathBuf(40000);
+      if (GetModuleFileNameW(
+            GetModuleHandle(nullptr), pathBuf.data(), static_cast<DWORD>(pathBuf.size())))
+      {
+        std::string progPath =
+          vtksys::SystemTools::GetProgramPath(vtksys::Encoding::ToNarrow(pathBuf.data()));
+        fullPath = vtksys::SystemTools::ConvertToWindowsOutputPath(progPath) + "\\";
+      }
+#endif
+
+      // construct the library file name from the plugin name
+      fullPath += vtksys::DynamicLoader::LibPrefix();
+      fullPath += "f3d-plugin-";
+      fullPath += path;
+      fullPath += vtksys::DynamicLoader::LibExtension();
+
+      handle = vtksys::DynamicLoader::OpenLibrary(fullPath);
+      if (!handle)
+      {
+        throw engine::plugin_exception(
+          "Cannot open the library \"" + path + "\": " + vtksys::DynamicLoader::LastError());
+      }
+    }
+
+    init_plugin = reinterpret_cast<factory::plugin_initializer_t>(
+      vtksys::DynamicLoader::GetSymbolAddress(handle, "init_plugin"));
+    if (init_plugin == nullptr)
+    {
+      throw engine::plugin_exception("Cannot find init_plugin symbol in library \"" + path +
+        "\": " + vtksys::DynamicLoader::LastError());
+    }
```

Comment:
add SDK test coevering this

Reply:
if possiblem not sure it is.

Reply:
I build all plugins as static libraries right now, so this code path cannot be tested. I'll consider building shared plugins, but it was causing issues in the CI for Windows and Mac.  
I can build shared plugins for Linux only though.

Reply:
The plugins are not found, even with `LD_LOAD_LIBRARY` properly set

Reply:
Is this CI specific or not ? If not, this is a big issue.

Reply:
Yes it's CI specific, I tested the feature on the 3 OS locally without any issue

Reply:
Ok

Reply:
ok, adding a linux only test is ok, fixing it is better.

Reply:
Actually, no I can't. `f3dPlugin.cmake` checks if VTK is built as a static library, and refuse to build plugins if they are not static.  
The idea is to avoid duplication of VTK in each plugin.  
Since VTK is built as a static library in the CI, plugins have to be built as static libraries too.

Reply:
got it, this will be tested once CI is redone without static.